chain/epoch-manager/src/lib.rs

```diff
@@ -403,40 +407,30 @@ impl EpochManager {
         // Later when we perform the check to kick out validators, we don't kick out validators in
         // exempted_validators.
         let mut exempted_validators = HashSet::new();
-        if checked_feature!("stable", MaxKickoutStake, epoch_info.protocol_version()) {
+        if ProtocolFeature::MaxKickoutStake.enabled(epoch_info.protocol_version()) {
             let min_keep_stake = total_stake * (exempt_perc as u128) / 100;
-            let mut sorted_validators = validator_block_chunk_stats
-                .iter()
-                .map(|(account, stats)| {
-                    let production_ratio =
-                        if stats.block_stats.expected == 0 && stats.chunk_stats.expected() == 0 {
-                            Rational64::from_integer(1)
-                        } else if stats.block_stats.expected == 0 {
-                            Rational64::new(
-                                stats.chunk_stats.produced() as i64,
-                                stats.chunk_stats.expected() as i64,
-                            )
-                        } else if stats.chunk_stats.expected() == 0 {
-                            Rational64::new(
-                                stats.block_stats.produced as i64,
-                                stats.block_stats.expected as i64,
-                            )
-                        } else {
-                            (Rational64::new(
-                                stats.chunk_stats.produced() as i64,
-                                stats.chunk_stats.expected() as i64,
-                            ) + Rational64::new(
-                                stats.block_stats.produced as i64,
-                                stats.block_stats.expected as i64,
-                            )) / 2
-                        };
-                    (production_ratio, account)
-                })
-                .collect::<Vec<_>>();
-            sorted_validators.sort();
+            let sorted_accounts = if ProtocolFeature::ChunkEndorsementsInBlockHeader
```

Comment:
nit: I would keep ChunkEndorsementsInBlockHeader check on the upper level, because for future protocol versions only this check will matter, and the other code paths can be moved somewhere else and called `get_old_online_ratio...`, which improves readability.

Reply:
done, moved the collection of sorted accounts to the caller function, which makes this code clener.